server/src/main/java/org/elasticsearch/cluster/ClusterInfoSimulator.java

```diff
@@ -39,10 +39,10 @@ public void simulate(ShardRouting shard) {
             if (shard.relocatingNodeId() != null) {
                 // relocation
                 modifyDiskUsage(shard.relocatingNodeId(), getShardPath(shard.relocatingNodeId(), mostAvailableSpaceUsage), size);
-                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), leastAvailableSpaceUsage), -size);
+                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), mostAvailableSpaceUsage), -size);
             } else {
                 // new shard
-                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), leastAvailableSpaceUsage), -size);
+                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), mostAvailableSpaceUsage), -size);
```

Comment:
We change the `mostAvailableSpaceUsage` instead as this is the one used in `DiskThresholdDecider#canAllocate` and would prevent overflowing disk with too many shards in case of multiple data paths. This also means we might allocate to few shards until we have refreshed cluster info in a next reroute.

Reply:
I think this is a decent compromise that degrades desired balance slightly when using multiple data paths, requiring shards to actually move before and a few iterations before it settles. Perhaps you can add a comment to the code though on this (something like your comment plus a bit).